src/runtime/generator.cpp

```diff
@@ -60,6 +60,29 @@ class RegisterHelper {
     }
 };
 
+static void freeGeneratorStack(BoxedGenerator* g) {
+    if (g->stack_begin == NULL)
+        return;
+
+    available_addrs.push_back((uint64_t)g->stack_begin);
+    // Limit the number of generator stacks we keep around:
+    if (available_addrs.size() > 5) {
+        uint64_t addr = available_addrs.front();
+        available_addrs.pop_front();
+        int r = munmap((void*)(addr - MAX_STACK_SIZE), MAX_STACK_SIZE);
+        assert(r == 0);
+// also unmap the redzone
+#if STACK_GROWS_DOWN
+        r = munmap((void*)addr, STACK_REDZONE_SIZE);
+        assert(r == 0);
```

Comment:
hmm you should double-check my math, but I thought this shouldn't be necessary since the redzone starts at `addr - MAX_STACK_SIZE`?